src/Foundation/Application.php

```diff
@@ -182,7 +208,66 @@ public function setThemesPath($path)
      */
     public function tempPath()
     {
-        return $this->basePath.'/storage/temp';
+        return $this->tempPath ?: $this->buildPath('/storage/temp');
+    }
+
+    /**
+     * Set the temp path for the application.
+     *
+     * @return string
+     */
+    public function setTempPath($path)
+    {
+        $path = str_replace('/', DIRECTORY_SEPARATOR, $path);
```

Comment:
Is there something in the PathResolver @bennothommo @mjauvin that could replace this? It's a bit of a code smell to have a one-liner path cleaner piece of code copy-pasted across every method instead of using a standard method to handle the common goal of making sure the input will generate clean output or fail appropriately regardless of what that input is.

Reply:
@LukeTowers I've replaced with @bennothommo 's PathResolver::standardize() method calls.